packages/gatsby-plugin-gatsby-cloud/src/ipc.js

```diff
@@ -0,0 +1,63 @@
+export function emitRoutes(routes) {
+  if (!process.send) {
+    return
+  }
+
+  process.send({
+    type: `LOG_ACTION`,
+    action: {
+      type: `CREATE_ROUTE`,
+      payload: {
+        routes,
+      },
+    },
+  })
+}
+
+export function emitRedirects(redirect) {
+  if (!process.send) {
+    return
+  }
+
+  process.send({
+    type: `LOG_ACTION`,
+    action: {
+      type: `CREATE_REDIRECT_ENTRY`,
+      payload: {
+        redirect,
+      },
```

Comment:
This drops one nested level of redirect as the type of the action already explain that this is a redirect
```suggestion
      payload: redirect,
```